### PR TITLE
Add support for contained levels in Javalab

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds support for contained levels in Javalab! If you seed this PR locally, you can see an example as the last Javalab allthethings level. I used studio.code.org/s/csd3-2021/lessons/5/levels/1?no_redirect=true as my non-Javalab example, if that helps anyone get a feel for how this works in other labs.

Adds the field to the level edit page:
Screen Shot 2021-07-30 at 10 00 28 AM

Getting the contained level to show up on the Javalab level, in view-only mode, didn't need a code change:
Screen Shot 2021-07-30 at 9 53 14 AM

The bulk of the non-test changes in this PR actually hook up the end-to-end experience, including:
- Preventing the student from changing their answer once `Run` is pressed
- Preventing the student from moving on until entering an answer
- Recording the student's answer when `Run` is pressed
- Allowing the student to "finish" the level once their answer is recorded

https://user-images.githubusercontent.com/46464143/127686455-d1592432-5d40-4467-888c-f722fa73560f.mov

I added a UI test for this feature. The existing contained level tests are all eyes tests, which I'm not completely sure are necessary here but I kept with that pattern.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/javalabRedux.js
Comment (bethanyaconnor) [inline apps/src/javalab/javalabRedux.js:143]: I don't love this pattern of using redux to set whether `Finish` can be pressed or not but I couldn't think of a better way. Definitely open to suggestions here.

File: dashboard/config/scripts/allthethings.script
Comment (molly-moen) [inline dashboard/config/scripts/allthethings.script]: should these non-javalab levels have been swapped?

File: dashboard/config/scripts_json/allthethings.script_json
Comment (molly-moen) [inline dashboard/config/scripts_json/allthethings.script_json]: double checking: are all the changes in this file intentional?

**Review by molly-moen — APPROVED** [review]
This looks great! Just a couple sanity check questions on the level changes

File: dashboard/config/scripts_json/allthethings.script_json
Comment (bethanyaconnor) [reply dashboard/config/scripts_json/allthethings.script_json]: ah no. I think this is mostly noise but I'll clean up this diff before merging

**Comment by bencodeorg** [comment]
Do we have any Javalab UI tests yet? I thought I heard at some point that there was too much churn in the UI at the moment (eg, buttons all just moved to the bottom of the page since you started working on this) so we didn't want to add them and have a lot of eyes diffs?

**Comment by bethanyaconnor** [comment]
> there was too much churn in the UI at the moment

hmm that's a good point. A UI test makes sense to me for this feature as it's pretty cross-cutting. I can hold off if we think there will be too much noise, but it would leave this feature untested.

**Comment by jmkulwik** [comment]
> Do we have any Javalab UI tests yet? I thought I heard at some point that there was too much churn in the UI at the moment (eg, buttons all just moved to the bottom of the page since you started working on this) so we didn't want to add them and have a lot of eyes diffs?

That's a great question. At this point, let's go ahead and start adding UI/Eyes tests where appropriate. Our UI churn should decrease significantly from here out.

**Pull request merged**
